shared/engine/index.js

```diff
@@ -266,11 +290,27 @@ class Engine {
     } else {
       invoke()
     }
+
+    return sessionID
   }
 
   rpc (params) {
     const {method, param, incomingCallMap, callback} = params
-    this.rpc_unchecked(method, param, incomingCallMap, callback)
+    return this.rpc_unchecked(method, param, incomingCallMap, callback)
+  }
+
+  cancelRPC (sessionID) {
+    const response = this.sessionIDToResponse[sessionID]
+    if (response) {
+      if (response && response.error) {
```

Comment:
Really the check for response can be removed but we should differentiate
between it being in the map or not separately if it has the error callback

On Tuesday, February 9, 2016, Marco Munizaga <notifications@github.com>
wrote:

> In shared/engine/index.js
> <https://github.com/keybase/client/pull/1939#discussion_r52392536>:
>
> >    }
> >
> >    rpc (params) {
> >      const {method, param, incomingCallMap, callback} = params
> > -    this.rpc_unchecked(method, param, incomingCallMap, callback)
> > +    return this.rpc_unchecked(method, param, incomingCallMap, callback)
> > +  }
> > +
> > +  cancelRPC (sessionID) {
> > +    const response = this.sessionIDToResponse[sessionID]
> > +    if (response) {
> > +      if (response && response.error) {
>
> can we collapse these two if statements?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/1939/files#r52392536>.
>